parser-typechecker/src/Unison/Codebase/Watch.hs

```diff
@@ -124,7 +114,7 @@ watchDirectory dir allow = do
       event@(file, _) <- watcher
       when (allow file) $
         STM.atomically $ TQueue.enqueue queue event
-  pending <- newIORef =<< existingFiles
+  pending <- newIORef []
```

Comment:
It is also weird to me that this functionality was implemented here. I don't think a directory watcher should automatically trigger an event when you first start watching the directory - that feels like it belongs in another layer if that is the behavior being implemented. 

A directory watcher should just emit events when files are changed or touched anytime after the watching begins.